commands/slash/attendance.go

```diff
@@ -0,0 +1,129 @@
+package slash
+
+import (
+	"fmt"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/ritsec/ops-bot-iii/commands/slash/permission"
+	"github.com/ritsec/ops-bot-iii/data"
+	"github.com/ritsec/ops-bot-iii/ent/signin"
+	"github.com/ritsec/ops-bot-iii/logging"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+//Attendance slash command
+func Attendance() (*discordgo.ApplicationCommand, func(s *discordgo.Session, i *discordgo.InteractionCreate)) {
+	return &discordgo.ApplicationCommand {
+		Name: 		 			  "attendance",
+		Description: 			  "Get signin history",
+		DefaultMemberPermissions: &permission.Member,
+	},
+	func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+		span := tracer.StartSpan(
+			"commands.slash.attendance:Attendance",
+			tracer.ResourceName("/attendance"),
+		)
+		defer span.Finish()
+
+		logging.Debug(s, "Attendance command received", i.Member.User, span)
+
+		err := s.InteractionRespond(
+			i.Interaction,
+			&discordgo.InteractionResponse {
+				Type: discordgo.InteractionResponseChannelMessageWithSource,
+				Data: &discordgo.InteractionResponseData {
+					Content: attendanceMessage(i.Member.User.ID, span.Context()),
+					Flags: discordgo.MessageFlagsEphemeral,
+				},
+			},
+		)
+		if err != nil {
+			logging.Error(s, err.Error(), i.Member.User, span)
+		} else {
+			logging.Debug(s, "Signin History Given", i.Member.User, span)
+		}
+	}
+}
+
+//returns the message sent to the user by the Attendance command
+func attendanceMessage(userID string, ctx ddtrace.SpanContext) (message string) {
+	span := tracer.StartSpan(
+		"commands.slash.attendance:attendanceMessage",
+		tracer.ResourceName("/attendance:attendanceMessage"),
+		tracer.ChildOf(ctx),
+	)
+	defer span.Finish()
+
+	message = "**Your Signins:**"
+	signinTypes := [...]string{
+		"General Meeting",
+		"Contagion",
+		"IR",
+		"Ops",
+		"Ops IG",
+		"Red Team",
+		"Red Team Recruiting",
+		"RVAPT",
+		"Reversing",
+		"Physical",
+		"Wireless",
+		"WiCyS",
+		"Vulnerability Research",
+		"Mentorship",
+		"Other",
+	}
+
+	totalSignins, err := data.Signin.GetSignins(userID, span.Context())
+	if err != nil {
+		logging.Error(nil, err.Error(), nil, span)
+		totalSignins = 0
+	}
+	message += fmt.Sprintf("\n\tTotal Signins: `%d`", totalSignins)
+
+	for _, signinType := range signinTypes {
+		var entSigninType signin.Type
+			switch signinType {
```

Comment:
@c0untingNumbers I'll pass it for this PR but just as a heads up, I noticed in a few other places we're hard coding the signin types as a string and putting it through a giant switch statement. I'll make an issue about it but could you look into refactoring this?

Ideally, we'd declare both in one place with a simple helper function which can then be called everywhere else (or something along these lines). Thanks!

Reply:
Yeah, good idea. Will put that up for an issue.